jans-cedarling/cedarling/src/authz/mod.rs

```diff
@@ -128,29 +116,145 @@ impl Authz {
                 LogType::Decision,
             )
             .set_auth_info(AuthorizationLogInfo {
-                action: request.action.clone(),
+                action: request.action,
                 context: request.context,
-                decision: decision.decision().into(),
-                principal: principal_workload_uid.to_string(),
-                diagnostics: decision.diagnostics().into(),
                 resource: resource_uid.to_string(),
+
+                person_principal: principal_user_entity_uid.to_string(),
+                workload_principal: principal_workload_uid.to_string(),
+
+                person_diagnostics: person_result.diagnostics().into(),
+                workload_diagnostics: workload_result.diagnostics().into(),
+
+                person_decision: person_result.decision().into(),
+                workload_decision: workload_result.decision().into(),
             })
-            .set_message("Result of authorize with resource as workload entity".to_string()),
+            .set_message("Result of authorize.".to_string()),
         );
 
-        Ok(AuthorizeResult { workload: decision })
+        Ok(AuthorizeResult {
+            workload: workload_result,
+            person: person_result,
+        })
+    }
+
+    /// Execute cedar policy is_authorized method to check
+    /// if allowed make request with given parameters
+    fn execute_authorize(
+        &self,
+        parameters: ExecuteAuthorizeParameters,
+    ) -> Result<cedar_policy::Response, cedar_policy::RequestValidationError> {
+        let request_principal_workload = cedar_policy::Request::new(
+            parameters.principal,
+            parameters.action,
+            parameters.resource,
+            parameters.context,
+            Some(&self.config.policy_store.schema.schema),
+        )?;
+
+        let response = self.authorizer.is_authorized(
+            &request_principal_workload,
+            &self.config.policy_store.policies,
+            parameters.entities,
+        );
+
+        Ok(response)
+    }
+
+    /// Create all [`Entity`]-s from [`Request`]
+    fn authorize_entities_data(
+        &self,
+        request: &Request,
+    ) -> Result<AuthorizeEntitiesData, AuthorizeError> {
+        let (access_token, id_token, user_info_token) = self
+            .config
+            .jwt_service
+            .decode_tokens::<AccessTokenData, IdTokenData, UserInfoTokenData>(
+                &request.access_token,
+                &request.id_token,
+                &request.userinfo_token,
+            )?;
+
+        // Using the builder at compile time ensures all entities are added correctly
```

Comment:
Using a single line of code doing too much work is concise and elegant. But we can put some more explanation in comments for helping developers and reviewers..

Reply:
Added comments